api/v1alpha1/asset_webhook.go

```diff
@@ -76,7 +76,15 @@ func (r *Asset) ValidateCreate() (admission.Warnings, error) {
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Asset) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
 	assetlog.Info("validate update", "name", r.Name)
-	return r.Validate()
+
+	warngins := admission.Warnings{}
+	if string(old.(*Asset).Spec.Type) != string(r.Spec.Type) {
+		warngins = append(warngins, "spec.type of a secret is immutable, delete the secret manually to get change affected")
```

Comment:
A warning is good, but won't it be great if we recreate the secret automatically?

Reply:
Yes, the only reason that I display the warning and not recreate secret is because I'm not familiar with controller runtime.
So I only made minimal changes.

Actually this warning didn't show up as expected too. I'm still figuring why.

Reply:
I'm still drafting this PR. I open this PR is for my team mate to check what have been change in my experiment image. That's why I set the PR base in my fork.

Reply:
OK, I've read the document and got a idea about how implement it. Maybe I will do it tomorrow after work.